### PR TITLE
Users Flow [Edit/Delete]

### PR DESCRIPTION
**Pull request by hericlesme**
## Description

Roles 
- New UI for Roles
- Edit and Delete Roles
- Role View (`users/roles/id`)

Groups
- Edit and Delete Groups
- Group View (`users/groups/id`)

Permissions
- It is now a object in `@/lib/permissions`

Modal
- All Dialogs updated to use `@reach/dialog`
- Added new Edit/Delete confirmation dialogs for roles and groups

API
- Split Users API Functions in 3 subsections (users, groups, roles)

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)

### CODE REVIEW

File: src/components/modals/groups/delete/index.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  isOpen: boolean
+  onClose: () => void
+  onConfirm: () => void
+}
+
+const DeleteGroupModal: FunctionComponent<IDeleteGroupModalProps> = ({id, isOpen, onClose, onConfirm}) => {
+  const handleSubmit = () => {
+    deleteGroup({id}).then(res => {
+      onConfirm()
+      console.log(res)
```
Comment (ariannmichael) [inline src/components/modals/groups/delete/index.tsx:17]: Should remove the consoles

File: src/components/modals/roles/delete/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  isOpen: boolean
+  onClose: () => void
+  onConfirm: () => void
+}
+
+const DeleteRoleModal: FunctionComponent<IDeleteRoleModalProps> = ({id, isOpen, onClose, onConfirm}) => {
+  const handleSubmit = () => {
+    deleteRole({id}).then(res => {
+      onConfirm()
+      console.log(res)
```
Comment (ariannmichael) [inline src/components/modals/roles/delete/index.tsx:17]: Should remove the consoles

File: src/components/modals/roles/edit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      canCreateUsers,
+      canCreateGroups,
+      canEditRoles,
+      canManageInfrastructure,
+      canUploadData
+    }
+
+    editRole(role).then(res => {
+      onConfirm()
+      console.log(res)
```
Comment (ariannmichael) [inline src/components/modals/roles/edit/index.tsx:51]: Should remove the consoles

**Review by ariannmichael — APPROVED** [review]

File: src/components/modals/roles/create/index.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+interface ICreateRoleModalProps {
+  isOpen: boolean
+  onClose: () => void
+  onConfirm: () => void
+}
+
+const CreateRoleModal: FunctionComponent<ICreateRoleModalProps> = ({isOpen, onClose, onConfirm}) => {
+  const [name, setName] = useState('')
+  const [canTriageRequests, setCanTriageRequests] = useState(false)
```
Comment (tcp) [inline src/components/modals/roles/create/index.tsx:15]: Simplify - we have 8 different `useState` here; seems like one object would do the trick.

File: src/components/modals/roles/create/index.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+              autoComplete="off"
+              className="shadow appearance-none border rounded w-full py-2 mb-4 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline border-blue-300 "
+              placeholder="User"
+              onChange={e => setName(e.target.value)}
+            />
+          </div>
+          <span className="text-sm block font-bold">Permissions</span>
+          {permissions.map(({id, value, setter}) => (
+            <div key={id} className="mt-2">
+              {/* eslint-disable-next-line jsx-a11y/no-onchange */}
```
Comment (tcp) [inline src/components/modals/roles/create/index.tsx:91]: Generally, try not to disable ESLint, especially a11y; why did you disable it here?

File: src/components/modals/roles/edit/index.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import {ButtonRound} from '@/components/lib'
```
Comment (tcp) [inline src/components/modals/roles/edit/index.tsx:1]: All these files are so similar, good opp for code reuse.

File: src/components/modals/roles/edit/index.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+import {FunctionComponent, useState} from 'react'
+import {Modal} from '../../modal'
+
+interface IEditRoleModalProps {
+  isOpen: boolean
+  onClose: () => void
+  onConfirm: () => void
+  id: string
+  name: string
+  canTriageRequests: boolean
```
Comment (tcp) [inline src/components/modals/roles/edit/index.tsx:13]: Use template literals to type the permissions and extend that to the roles.

File: src/types/api-responses.ts [hunk 8]
```diff
@@ -25,3 +23,19 @@ export interface ICreateUserResponse {
 export interface ICreateGroupResponse {
   message: string
 }
+
+export interface ICreateRoleResponse {
```
Comment (tcp) [inline src/types/api-responses.ts:27]: They're all the same - use a single type for this response.

**Review by tcp — APPROVED** [review]
Hey @hericlesme! I'll approve. However, there's still room to improve the UI and I'd like to see some other changes in structure and data fetching. I've created new issues specifically for handling this and we'll handle it in the next couple of days.

Thank you!

**Pull request merged**